h/static/scripts/hooks/use-group-annotations.tsx

```diff
@@ -178,13 +224,15 @@ export function useGroupAnnotations({
 
   // When annotations is not defined, trigger first load
   useEffect(() => {
-    requestController.current = new AbortController();
     if (annotations === undefined) {
       loadAnnotationsForCurrentPage();
     }
-    return () => requestController.current?.abort();
   }, [annotations, loadAnnotationsForCurrentPage]);
 
+  useEffect(() => {
+    return () => requestController.current?.abort();
+  }, []);
```

Comment:
I'm not completely sure if this is actually useful.

We are currently creating a new AbortController every time a request is triggered, but this just cancels the last one, even though multiple requests could be in flight "at this point".

Additionally, these requests can be triggered by multiple conditions, and not in all of them it makes sense to abort the previous request.

Reply:
Yeah, so we should track a list of requests, not the "last" one. That makes sense.